Error in `FetchContent_Declare` for `json`

With the following system configuration:

  - System: Perlmutter
  - OS: SLES 15.5
  - `cmake`: 3.20.4
  - `qiskit-c-api-demo`: 0c81116ee24e4c4624c978778b0b8d240a28b19f

issuing `cmake <...> ..` leads to the following error:
```
CMake Error at /usr/share/cmake/Modules/ExternalProject.cmake:2773 (message):
  At least one entry of URL is a path (invalid in a list)
Call Stack (most recent call first):
  /usr/share/cmake/Modules/ExternalProject.cmake:3683 (_ep_add_download_command)
  CMakeLists.txt:15 (ExternalProject_Add)


-- Configuring incomplete, errors occurred!
See also "/global/u1/v/vrpascuz/devel/qiskit-c-api-demo/build/_deps/json-subbuild/CMakeFiles/CMakeOutput.log".

CMake Error at /usr/share/cmake/Modules/FetchContent.cmake:1000 (message):
  CMake step for json failed: 1
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FetchContent.cmake:1141:EVAL:2 (__FetchContent_directPopulate)
  /usr/share/cmake/Modules/FetchContent.cmake:1141 (cmake_language)
  /usr/share/cmake/Modules/FetchContent.cmake:1184 (FetchContent_Populate)
  CMakeLists.txt:50 (FetchContent_MakeAvailable)


-- Configuring incomplete, errors occurred!
See also "/global/homes/v/vrpascuz/devel/qiskit-c-api-demo/build/CMakeFiles/CMakeOutput.log".
```

This is a result of the order of arguments:
https://github.com/qiskit-community/qiskit-c-api-demo/blob/0c81116ee24e4c4624c978778b0b8d240a28b19f/CMakeLists.txt#L44

Specifically, `DOWNLOAD_EXTRACT_TIMESTAMP TRUE` strings are interpreted as a URLs and therefore the call fails. Moving `DOWNLOAD_EXTRACT_TIMESTAMP TRUE` _before_ `URL` fixes the error.

Anyone else experience this behavior? Happy to submit a PR, if helpful.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -41,7 +41,11 @@ add_library(qiskit_c_api STATIC IMPORTED)
 
 # nlohmann_json
 include(FetchContent)
-FetchContent_Declare(json URL https://github.com/nlohmann/json/releases/download/v3.12.0/json.tar.xz DOWNLOAD_EXTRACT_TIMESTAMP TRUE)
+FetchContent_Declare(
+    json
+    DOWNLOAD_EXTRACT_TIMESTAMP TRUE
+    URL https://github.com/nlohmann/json/releases/download/v3.12.0/json.tar.xz
+)
 FetchContent_MakeAvailable(json)
 
 # qiskit library

